torchaudio/__init__.py

```diff
@@ -29,6 +29,7 @@
 
 if _mod_utils.is_module_available('torchaudio._torchaudio'):
     from . import _torchaudio
+    initialize_sox()
```

Comment:
nit: We were not initializing sox at import time. I don't see a downside, so I'll just leave a note.

Reply:
Yup. 
With `is_module_available` and less function definition in `__init__.py`, we can actually do what `__init__.py` is supposed to do. Initialization :) .